webapp/src/features/adminEditIntersection/AdminEditIntersection.tsx

```diff
@@ -0,0 +1,334 @@
+import React, { useEffect, useState } from 'react'
+import { Form } from 'react-bootstrap'
+import { useForm } from 'react-hook-form'
+import { ErrorMessage } from '@hookform/error-message'
+import { Multiselect } from 'react-widgets'
+import {
+  selectApiData,
+  selectOrganizations,
+  selectSelectedOrganizations,
+  selectRsus,
+  selectSelectedRsus,
+  selectSubmitAttempt,
+
+  // actions
+  getIntersectionInfo,
+  submitForm,
+  setSelectedOrganizations,
+  setSelectedRsus,
+} from './adminEditIntersectionSlice'
+import { useSelector, useDispatch } from 'react-redux'
+
+import '../adminIntersectionTab/Admin.css'
+import { AnyAction, ThunkDispatch } from '@reduxjs/toolkit'
+import { RootState } from '../../store'
+import { AdminIntersection } from '../../models/Intersection'
+import { Link, useNavigate, useParams } from 'react-router-dom'
+import { selectTableData, updateTableData } from '../adminIntersectionTab/adminIntersectionTabSlice'
+import { Dialog, DialogActions, DialogContent, DialogTitle, Typography } from '@mui/material'
+import toast from 'react-hot-toast'
+
+export type AdminEditIntersectionFormType = {
+  intersection_id: string
+  ref_pt: {
+    latitude: string
+    longitude: string
+  }
+  bbox?: {
+    latitude1: string
+    longitude1: string
+    latitude2: string
+    longitude2: string
+  }
+  intersection_name?: string
+  origin_ip?: string
+  organizations: string[]
+  organizations_to_add: string[]
+  organizations_to_remove: string[]
+  rsus: string[]
+  rsus_to_add: string[]
+  rsus_to_remove: string[]
+}
+
+const AdminEditIntersection = () => {
+  const dispatch: ThunkDispatch<RootState, void, AnyAction> = useDispatch()
+  const apiData = useSelector(selectApiData)
+  const organizations = useSelector(selectOrganizations)
+  const selectedOrganizations = useSelector(selectSelectedOrganizations)
+  const rsus = useSelector(selectRsus)
+  const selectedRsus = useSelector(selectSelectedRsus)
+  const submitAttempt = useSelector(selectSubmitAttempt)
+  const intersectionTableData = useSelector(selectTableData)
+
+  const [open, setOpen] = useState(true)
+  const navigate = useNavigate()
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    setValue,
+  } = useForm<AdminEditIntersectionFormType>({
+    defaultValues: {
+      intersection_id: '',
```

Comment:
Is there much value to allowing intersection ID to be seen on the modification when the API doesn't actually allow for this to be modified?

Reply:
The intersection_id corresponds to the intersection_number in the sql table. The intersection_id (intersection_number in sql) I believe should be able to be changed

Reply:
Added the ability to modify the intersection ID